tgui/packages/tgui/interfaces/LibraryConsole.js

```diff
@@ -0,0 +1,988 @@
+          minimum_page_count={1}
+          page_count={inventory_page_count}
+          current_page={inventory_page}
+          call_on_change={(value) => act('switch_inventory_page', {
+            page: value,
+          })} />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+export const InventoryDetails = (props, context) => {
+  const { act, data } = useBackend(context);
+  const inventory = flow([
+    map((book, i) => ({
+      ...book,
+      // Generate a unique id
+      key: i,
+    })),
+    sortBy(book => book.key),
+  ])(data.inventory);
+  return (
+    <Section>
+      <Table>
+        <Table.Row>
+          <Table.Cell
+            fontSize={1.5}>
+            Remove
+          </Table.Cell>
+          <Table.Cell
+            fontSize={1.5}>
+            Title
+          </Table.Cell>
+          <Table.Cell
+            fontSize={1.5}>
+            Author
+          </Table.Cell>
+        </Table.Row>
+        {inventory.map(book => (
+          <Table.Row key={book.key}>
+            <Table.Cell>
+              <Button
+                color="bad"
+                onClick={() => act('inventory_remove', {
+                  book_id: book.ref,
+                })}
+                icon="times">
+                Clear Record
+              </Button>
+            </Table.Cell>
+            <Table.Cell>
+              {book.title}
+            </Table.Cell>
+            <Table.Cell>
+              {book.author}
+            </Table.Cell>
+          </Table.Row>
+        ))}
+      </Table>
+    </Section>
+  );
+};
+
+export const Checkout = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    checkout_page,
+    checkout_page_count,
+  } = data;
+
+  const [checkoutBook, setCheckoutBook] = useLocalState(
+    context, "CheckoutBook", false);
+  return (
+    <Stack vertical height="100%" justify="space-between">
+      <Stack.Item grow>
+        <Stack vertical height="100%">
+          <Stack.Item grow>
+            <ScrollableSection
+              header="Checked Out Books"
+              contents={<CheckoutEntries />} />
+          </Stack.Item>
+          <Stack.Item align="center">
+            <PageSelect
+              minimum_page_count={1}
+              page_count={checkout_page_count}
+              current_page={checkout_page}
+              call_on_change={(value) => act('switch_checkout_page', {
+                page: value,
+              })} />
+          </Stack.Item>
+        </Stack>
+      </Stack.Item>
+      <Stack.Item>
+        <Button
+          fluid
+          icon="barcode"
+          content="Check-Out Book"
+          fontSize="20px"
+          onClick={() => setCheckoutBook(true)} />
+      </Stack.Item>
+      {!!checkoutBook && <CheckoutModal />}
+    </Stack>
+  );
+};
+
+export const CheckoutEntries = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    checkouts,
+    has_checkout,
+  } = data;
+
+  if (!has_checkout) {
+    return;
```

Comment:
Returning `null` is a good practice, since `undefined` normally means that you forgot to return something in the component. I know this because React catches these types of inaccuracies, and Inferno doesn't. Not very critical.

```suggestion
    return null;
```